sys_util/src/lib.rs

```diff
@@ -18,9 +18,36 @@ mod signal;
 mod struct_util;
 mod terminal;
 
-pub use errno::{errno_result, Error, Result};
+pub use errno::Error;
 pub use eventfd::*;
 pub use ioctl::*;
 pub use signal::*;
-pub use struct_util::*;
+pub use struct_util::{read_struct, read_struct_slice};
 pub use terminal::*;
+
+#[doc(hidden)]
+pub trait IsMinusOne {
+    fn is_minus_one(&self) -> bool;
+}
+
+macro_rules! impl_is_minus_one {
+    ($($t:ident)*) => ($(impl IsMinusOne for $t {
+        fn is_minus_one(&self) -> bool {
+            *self == -1
+        }
+    })*)
+}
+
+impl_is_minus_one! { i8 i16 i32 i64 isize }
+
+pub fn cvt<T: IsMinusOne>(t: T) -> std::io::Result<T> {
```

Comment:
Believe me that I have tried to figure that out myself and failed :laughing: 

I did a bit of research on what is the standard way to handle syscalls return values and errors and found that many crates including `libstd` use functions with more or less the same implementation as the one in this PR and call them ... `cvt`. Ctrl+F for `cvt` here for example: https://github.com/rust-lang/rust/blob/master/src/libstd/sys/unix/os.rs

Now we can either call it `cvt` too, or try to come up with a better name (`syscall_result`, `interpret_exit_code`, `resultify`, `interpret_result`, etc).

Added documentation as requested.

Reply:
Figured one out: cvt apparently stands for convert. That's still a very awkward name. It might be that there are already some wrappers over this `cvt` functionality that we could use from the standard library.

Reply:
Unfortunately I couldn't find any. Even the definition of the function itself is in a private module.